crates/tags/src/lib.rs

```diff
@@ -31,20 +30,20 @@ impl<T: Clone + Storage> Service<T> {
         self.0.tags().await.map(Json)
     }
 
-    async fn delete(&self, tag: Tag) -> Result<impl IntoResponse> {
+    async fn delete(&self, tag: TagName) -> Result<impl IntoResponse> {
         self.0.del(tag).await
     }
 
-    async fn head(&self, tag: Tag) -> Result<impl IntoResponse> {
+    async fn head(&self, tag: TagName) -> Result<impl IntoResponse> {
         self.0.get(tag).await.map(|_| ())
     }
 
-    async fn get(&self, tag: Tag) -> Result<impl IntoResponse> {
+    async fn get(&self, tag: TagName) -> Result<impl IntoResponse> {
         self.0.get(tag).await.map(Json)
     }
 
-    async fn put(&self, tag: Tag, hash: Hash) -> Result<impl IntoResponse> {
-        self.0.put(tag, hash).await
+    async fn put(&self, tag: TagName, data: TagValue) -> Result<impl IntoResponse> {
```

Comment:
Since we're getting rid of particular tag name format, can we just use `String` as name types here?

Reply:
No, because `String` doesn't implement `FromRequest`. I've already tried.

Reply:
This is a conversation from a few days ago where I had issues using basic Rust types with the server code, because of types not implementing `FromRequest`. It causes the handler function to have problems with the closures. https://chat.enarx.dev/channel/development?msg=q7i5G5xDpXvkWFSTY

Reply:
Right, so to fix it you have to specify the type (`TagName`) in the closure on L63 and then call this function with unwrapped value

Something like this:
```
(.., Method::Put) => (|t: TagName, d: TagValue| self.put(t.0, d)).handle(req).await,
```